src/System.Linq.Expressions/src/System/Linq/Expressions/NewExpression.cs

```diff
@@ -162,8 +162,8 @@ public static NewExpression New(ConstructorInfo constructor, params Expression[]
         /// <returns>A <see cref="NewExpression"/> that has the <see cref="NodeType"/> property equal to <see cref="P:New"/> and the <see cref="P:Constructor"/> and <see cref="P:Arguments"/> properties set to the specified value.</returns>
         public static NewExpression New(ConstructorInfo constructor, IEnumerable<Expression> arguments)
         {
-            ContractUtils.RequiresNotNull(constructor, "constructor");
-            ContractUtils.RequiresNotNull(constructor.DeclaringType, "constructor.DeclaringType");
+            ContractUtils.RequiresNotNull(constructor, nameof(constructor));
+            ContractUtils.RequiresNotNull(constructor.DeclaringType, nameof(constructor) + "." + nameof(constructor.DeclaringType));
```

Comment:
Though that string is `"DeclaringType"` rather than `"constructor.DeclaringType"` so it's no longer as informative. (The practice in Expressions of such of dotted argument name values seems unusual, and possibly wrong, to me, but maybe it's common and I just haven't seen it much).